spellbook/models/looksrare/ethereum/looksrare_ethereum_events.sql

```diff
@@ -0,0 +1,157 @@
+{{ config(
+        alias ='events',
+        materialized ='incremental',
+        file_format ='delta',
+        incremental_strategy='merge',
+        unique_key='unique_trade_id'
+        )
+}}
+
+WITH looks_rare AS (
+        SELECT 
+        ask.evt_block_time AS block_time,
+        ask.tokenId::string AS token_id,
+        ask.amount AS number_of_items,
+        taker AS seller,
+        maker AS buyer,
+        price AS price,
+        roy.amount AS royalty_fee,
+        roy.royaltyRecipient AS royalty_fee_receive_address,
+        roy.currency AS royalty_fee_currency_symbol,
+        CASE -- REPLACE `ETH` WITH `WETH` for ERC20 lookup later
+            WHEN ask.currency = '0x0000000000000000000000000000000000000000' THEN '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
+            ELSE ask.currency
+        END AS currency_contract,
+        ask.currency AS currency_contract_original,
+        ask.collection AS nft_contract_address,
+        ask.contract_address AS contract_address,
+        ask.evt_tx_hash AS tx_hash,
+        ask.evt_block_number AS block_number,
+        ask.evt_index AS evt_index,
+        roy.evt_index as roy_event_index,
+        CASE -- CATEGORIZE Collection Wide Offers Accepted 
+            WHEN strategy = '0x86f909f70813cdb1bc733f4d97dc6b03b8e7e8f3' THEN 'Collection Offer Accepted'
+            ELSE 'Offer Accepted' 
+            END AS category    
+    FROM {{ source('looksrare_ethereum','looksrareexchange_evt_takerask') }} ask
```

Comment:
in the config, it's set to incremental & the sources file for looksrare seems to call out freshness on block time column, but how does dbt know to read only incremental data rather than the entire source dataset? is that automatically applied, or should it be explicitly written in this code?

Reply:
if it needs to be written out, i think all table reads and joins to larger tables (i.e. ethereum transactions outside this CTE) should include filters on dates, if possible?

Reply:
- You're right it should be explicitly specified (otherwise it does recompute everything every time it runs!), I added this to the code:
{% if is_incremental() %}
WHERE looks_rare.block_time > now() - interval 2 days
{% endif %} 

- Good thinking, I added a time filter on both erc721/1155 tables + ethereum.transactions, starting when the LooksRare platform launched (which I set to january 1st 2022)

Reply:
re: time filter on both erc721/1155 tables + ethereum.transactions -- i think we should restrict the joins to same filter as looks_rare subquery (i.e. > now() - interval 2 days). any reason the filters can't match between the two? 

over time, > '2022-01-01' can still take more time than necessary to process rows we don't need in current execution